### PR TITLE
rename study

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: AbdelHedhili <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

File: src/components/study-manager.js [hunk 1]
```diff
@@ -253,6 +294,23 @@ const StudyCard = ({study, onClick}) => {
                     </Button>
                 </DialogActions>
             </Dialog>
+            <Dialog open={openRenameDialog} onClose={handleCloseRenameDialog} aria-labelledby="form-dialog-title">
```
Comment (jonenst) [inline src/components/study-manager.js:297]: the aria-labelledby is the id of the DialogTitle and should be unique

File: src/components/study-manager.js [hunk 2]
```diff
@@ -89,7 +92,9 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const StudyCard = ({study, onClick}) => {
-    const [open, setOpen] = React.useState(false);
+    const [openDeleteDialog, setOpenDelete] = React.useState(false);
+    const [openRenameDialog, setOpenRename]  = React.useState(false);
+    const [newStudyNameValue, setNewStudyName] = React.useState(study.studyName);
```
Comment (jonenst) [inline src/components/study-manager.js:97]: I think it's weird to put this state in the studycard. Maybe extract it to another component ? The alternative is to use uncontrolled forms but that's a bit of an antipattern in react

Comment (jonenst) [reply src/components/study-manager.js:97]: To be more precise, can we use more components to restrict the scope of relayouting (at each keystroke in input in the dialog) ?

ie move the studyName in something like:
```
const renameDialog = () => {
     const [newStudyNameValue, setNewStudyName] = React.useState(study.studyName); 
}
```

Comment (AbdelHedhili) [reply src/components/study-manager.js]: @jonenst I made the change for renameDialog, is that what you expected ?

File: src/components/study-manager.js [hunk 3]
```diff
@@ -253,10 +288,48 @@ const StudyCard = ({study, onClick}) => {
                     </Button>
                 </DialogActions>
             </Dialog>
+            <RenameDialog study={study}
```
Comment (jonenst) [inline src/components/study-manager.js:291]: pass studyname directly to props

File: src/components/study-manager.js [hunk 4]
```diff
@@ -253,10 +288,48 @@ const StudyCard = ({study, onClick}) => {
...
+        console.log(newStudyNameValue);
+        props.handleConfirm(newStudyNameValue);
+    };
+
+    return (
+        <Dialog open={props.openRenameDialog} onClose={props.handleCloseDialog} aria-labelledby="dialog-title-rename">
+            <DialogTitle id="dialog-title-rename"><FormattedMessage id="renameStudy"/></DialogTitle>
+            <DialogContent>
+                <InputLabel htmlFor="newStudyName"><FormattedMessage id="renameStudyMsg"/></InputLabel>
+                <TextField id="newStudyName" defaultValue={props.study.studyName} required={true} onChange={updateStudyNameValue}></TextField>
```
Comment (jonenst) [inline src/components/study-manager.js:319]: We should either use value and a state (controlled component) or defautlValue and  a ref (uncontrolled component). Controlled components are the officially recommended solution

**Pull request merged**